platform/ext/target/nordic_nrf/common/native_drivers/spu.c

```diff
@@ -151,3 +151,47 @@ void spu_regions_flash_config_non_secure_callable(uint32_t start_addr,
 		FLASH_NSC_REGION_FROM_ADDR(start_addr),
 		1 /* Lock */);
 }
+
+void spu_peripheral_config_secure(uint32_t periph_base_addr, bool periph_lock)
+{
+	/* Determine peripheral ID */
+	const uint8_t periph_id = NRFX_PERIPHERAL_ID_GET(periph_base_addr);
+
+	/* ASSERT checking that this is not an explicit Non-Secure peripheral */
+	NRFX_ASSERT((NRF_SPU->PERIPHID[periph_id].PERM &
+		SPU_PERIPHID_PERM_SECUREMAPPING_Msk) !=
+		(SPU_PERIPHID_PERM_SECUREMAPPING_NonSecure <<
+			SPU_PERIPHID_PERM_SECUREMAPPING_Pos));
+
+	nrf_spu_peripheral_set(NRF_SPU, periph_id,
+		1 /* Secure */,
+		1 /* Secure DMA */,
+		lock /* Lock */);
+}
+
+void spu_peripheral_config_non_secure(uint32_t periph_base_addr, bool periph_lock)
+{
+	/* Determine peripheral ID */
+	const uint8_t periph_id = NRFX_PERIPHERAL_ID_GET(periph_base_addr);
+
+	/* ASSERT checking that this is not an explicit Secure peripheral */
+	NRFX_ASSERT((NRF_SPU->PERIPHID[periph_id].PERM &
+		SPU_PERIPHID_PERM_SECUREMAPPING_Msk) !=
+		(SPU_PERIPHID_PERM_SECUREMAPPING_Secure <<
+			SPU_PERIPHID_PERM_SECUREMAPPING_Pos));
+
+	nrf_spu_peripheral_set(NRF_SPU, periph_id,
+		0 /* Non-Secure */,
+		0 /* Non-Secure DMA */,
+		lock /* Lock */);
+}
+
+void spu_dppi_config_non_secure(bool dppi_lock)
+{
+	nrf_spu_dppi_config_set(NRF_SPU, 0, 0xFFFFFFFF, dppi_lock);
+}
+
+void spu_gpio_config_non_secure(bool gpio_lock)
+{
+	nrf_spu_gpio_config_set(NRF_SPU, 0, 0xFFFFFFFF, gpio_lock);
```

Comment:
And what with P1 on nRF5340?

Reply:
Good point, this is how we have this at the moment in downstream SPM. 
@oyvindronningstad FYI: maybe we should leave P1 to secure domain in general??

Reply:
It might be clearer if the ports are configurable, and everything is NS by default. Either that, or clearly explain it in the header file.

Reply:
Why not take the port number as the parameter of `spu_gpio_config_non_secure()`?
BTW this function, as well as `spu_dppi_config_non_secure`, could have `__STATIC_INLINE` implementation in `spu.h`. The generated code would be probably shorter.

Reply:
Yeap, these are all good suggestions. So I will:
- I will pass the port number as parameter to the public API function
- I will set P1 as well to NS  (by default) during startup configuration
- I 'll use `static inline` for these functions and "implement" the in the header

Reply:
Updated now, @oyvindronningstad @anangl